src/main/java/com/microsoft/sqlserver/jdbc/Util.java

```diff
@@ -34,6 +36,8 @@ final class Util {
 
     final static String ACTIVITY_ID_TRACE_PROPERTY = "com.microsoft.sqlserver.jdbc.traceactivity";
 
+    final static Pattern SERVER_NAME_PATTERN = Pattern.compile("((?=^.{1,253}$)(^(((?!-)[a-zA-Z0-9-'\"_]{1,}(?<!-))|((?!-)[a-zA-Z0-9-'\"_]{1,}(?<!-)\\.)+[a-zA-Z]{2,})$))|(\\d+\\.?)+");
```

Comment:
Can we include a comment above this line that describes what rules the regex is trying to enforce?

My concern is we might end up breaking something like [IDN support](https://en.wikipedia.org/wiki/Internationalized_domain_name). Could we simplify this check by simply checking for an `=` in the serverName field, which would indicate the user is really trying to specify a property and is the potentially common error we are guarding against?

Reply:
yeah it's not too bad since we only parse this once but regex is really slow. As for IDN, we have tests for that and also we have fuzz tests for IDN you can run